feat(GiniHealthSDK): Payment Feature Info Screen IPC-102

razvancapra opened the pull request:
Payment Feature Info Screen about the payment feature for each payable invoice
Also cotains fixing rounding and bordering of bank icons: https://ginis.atlassian.net/browse/IPC-172

IPC-102 
IPC-172

ValentinaIancu-Gini commented:
@razvancapra Thanks for the work done!

I left a few suggestions and clarifications.

However, I found some small issues besides those.

1. After expending to see the answers to the questions, in a random order, and mostly if I start with the last one and after that others, the answer is not entirely visible due to the reusability feature. Some cells may be reused with a smaller size than the size that is needed.
Please see the attached screenshots.
<img width="815" alt="Screenshot 2024-03-08 at 19 09 24" src="https://github.com/gini/gini-mobile-ios/assets/132361182/eff8f555-894b-4afc-ab92-dde61fa9eb64">

<img width="810" alt="Screenshot 2024-03-08 at 19 10 00" src="https://github.com/gini/gini-mobile-ios/assets/132361182/c3e1bd22-1a6d-4c57-b7bd-1badbc9675dc">


2. Another issue, which might not be an issue but I want to confirm it, is when I select to pay the invoice. On one device I received an error even if I installed both apps needed for this. And on another device, the list of providers seemed too long and I wanted to scroll but I ended up selecting a new provider, also I was stuck in the payment screen. Please see the videos attached and if this is not an issue ignore my comment.


https://github.com/gini/gini-mobile-ios/assets/132361182/c4a043b2-9bb1-440e-bb82-1e1874b665c2


https://github.com/gini/gini-mobile-ios/assets/132361182/8d342971-51ba-4941-b501-d99cf222589e

3. Switching to German the `[LINK]` text is not replaced and `"Pay bills easily with the banking app."`
![Simulator Screenshot - iPhone SE (1st generation) - 2024-03-08 at 19 40 26](https://github.com/gini/gini-mobile-ios/assets/132361182/425711c7-4445-4ef3-888c-f2602d5388c9)


4. Tapping multiple times on the `+` icon to expand/collapse the question to see the answer the table view content offset is growing.
<img width="506" alt="Screenshot 2024-03-08 at 19 44 51" src="https://github.com/gini/gini-mobile-ios/assets/132361182/abb74044-fe79-4ff2-8ff3-88845611056e">


The last remark is that on the iPhone SE first generation, the `Back` button text is missing while switching to German. Maybe because is longer than in English: `Zurück` vs `Back`.


https://github.com/gini/gini-mobile-ios/assets/132361182/20bf0a87-e223-4604-b77d-dc5282cbb7b5

razvancapra commented on HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/PaymentInfoViewController.swift:
```diff
@@ -0,0 +1,349 @@
+        if #available(iOS 15.0, *) {
+            tableView.sectionHeaderTopPadding = 0
+        } else {
+            // Fallback on earlier versions
```
Since iOS 15, UITableView contains a new property called [sectionHeaderTopPadding](https://developer.apple.com/documentation/uikit/uitableview/3750914-sectionheadertoppadding) which specifies the amount of padding above each section header.

Yes, we need this. Otherwise we will have extra padding above each section from UITableView.

razvancapra commented on HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/PaymentInfoViewController.swift:
```diff
@@ -0,0 +1,349 @@
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        let cellCount = CGFloat(viewModel.paymentProviders.count)
```
I've changed and transformed to Double. We need this to calculate the total width of collectionview

zladzeyka reviewed (commented):
Great job @razvancapra 💪🏻
Left you a few tiny suggestions. 🙏

razvancapra commented on HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentReviewViewController.swift:
```diff
@@ -63,7 +63,7 @@ public final class PaymentReviewViewController: UIViewController, UIGestureRecog
         let vc = (UIStoryboard(name: "PaymentReview", bundle: giniHealthBundle())
             .instantiateViewController(withIdentifier: "paymentReviewViewController") as? PaymentReviewViewController)!
-        vc.model = PaymentReviewModel(with: giniHealth, document: document, extractions: extractions )
+        vc.model = PaymentReviewModel(with: giniHealth, document: document, extractions: extractions)
```
I just used the naming used on other initializers. I will change them all.

razvancapra commented:
@ValentinaIancu-Gini Regarding the 4 issues reported:
1. Fixed 👍 
2. The banks picker from Payment Review Screen is going to be remove in the next PR. The error shown is a server error received from API when requesting invoice extractions to review in Payment Review screen. Can't reproduce yet.
3. fixed 👍 
4. fixed 👍

zladzeyka reviewed (commented):
@razvancapra 
Looks good to me.
I've noticed a weird case, after expanding the last question and collapsing it back we've got the cell delimiter at the bottom.


https://github.com/gini/gini-mobile-ios/assets/3911816/cdcdbd34-bf9c-41e5-af1b-d30dbb0b7222


And one more tiny thing: When we have a long list of payment providers: the 1st element needs an inset it thinks.

![Simulator Screenshot - iPhone SE (3rd generation) - 2024-03-14 at 15 48 46](https://github.com/gini/gini-mobile-ios/assets/3911816/a6c60567-0af1-4b3a-aa29-683c73edd910)

zladzeyka reviewed (approved):
Thank you @razvancapra
